oak_attestation_verification/tests/endorsement_tests.rs

```diff
@@ -80,6 +90,50 @@ fn load_testdata() -> TestData {
     }
 }
 
+#[test]
+fn test_verify_binary_digest_same() {
+    let testdata = load_testdata();
+    let expected = create_hex_digest();
+    let result = verify_binary_digest(&testdata.endorsement, &expected);
+    assert!(result.is_ok_and(|m| m == MatchResult::SAME));
+}
+
+#[test]
+fn test_verify_binary_digest_different() {
+    let testdata = load_testdata();
+    let expected = HexDigest {
+        psha2: BINARY_DIGEST.to_owned(),
+        sha1: BINARY_DIGEST.to_owned(),
+        sha2_256: "00000bad000digestb91bd22ee4c976420f8f0c6a895fd083dcb0d0000000000".to_owned(),
+        sha2_512: BINARY_DIGEST.to_owned(),
+        sha3_512: BINARY_DIGEST.to_owned(),
+        sha3_384: BINARY_DIGEST.to_owned(),
+        sha3_256: BINARY_DIGEST.to_owned(),
+        sha3_224: BINARY_DIGEST.to_owned(),
+        sha2_384: BINARY_DIGEST.to_owned(),
+    };
+    let result = verify_binary_digest(&testdata.endorsement, &expected);
+    assert!(result.is_ok_and(|m| m == MatchResult::DIFFERENT));
+}
+
+#[test]
+fn test_verify_binary_digest_undecidable() {
+    let testdata = load_testdata();
+    let expected = HexDigest {
+        psha2: BINARY_DIGEST.to_owned(),
+        sha1: BINARY_DIGEST.to_owned(),
+        sha2_256: "".to_owned(),
+        sha2_512: BINARY_DIGEST.to_owned(),
+        sha3_512: BINARY_DIGEST.to_owned(),
+        sha3_384: BINARY_DIGEST.to_owned(),
+        sha3_256: BINARY_DIGEST.to_owned(),
+        sha3_224: BINARY_DIGEST.to_owned(),
+        sha2_384: BINARY_DIGEST.to_owned(),
+    };
+    let result = verify_binary_digest(&testdata.endorsement, &expected);
+    assert!(result.is_ok_and(|m| m == MatchResult::UNDECIDABLE));
```

Comment:
I can imagine that DIFFERENT and UNDECIDABLE will be treated similarly in many cases (e.g. bail! with a different message). But shouldn't the library/utility function distinguish this detail?

Reply:
In my mind if we can induce a total order (instead of just a [preorder](https://en.wikipedia.org/wiki/Preorder) or a partial order) on the lattice of digests, it seems a useful abstraction in practice, and it is a more familiar structure from a mathematical point of view. But I do see your point of view, having the explicit third state may have a semantic advantage to represent some categories of errors.